test/files/neg/t11618.check

```diff
@@ -0,0 +1,36 @@
+t11618.scala:9: warning: Implicit pattern definition binds no variables
+  implicit val _ = 42
+               ^
+t11618.scala:10: warning: Implicit pattern definition binds no variables
+  implicit val Some(_) = Option(42)
+                   ^
+t11618.scala:18: warning: Implicit pattern definition binds no variables
+    implicit val _ = i
+                 ^
+t11618.scala:19: warning: Implicit pattern definition binds no variables
+    implicit val Some(_) = Option(i)
+                     ^
+t11618.scala:24: warning: Implicit pattern definition binds no variables
+  implicit val Some(_) = Option(42)
+                   ^
+t11618.scala:7: warning: Pattern definition introduces Unit-valued member of C
+  val _ = 42
+      ^
+t11618.scala:8: warning: Pattern definition introduces Unit-valued member of C
+  val Some(_) = Option(42)
+          ^
```

Comment:
I don't think this should warn.  Could you include `val _: Unit = ...` val Some(_: Unit) = ...` variants too, please?  I find these useful as ways to signal (and catch regressions) that the RHS is there for its effect, as noted by the Unit result type.  It's not an error.  (Ideally, if it returned something different (a Boolean, a case class value, anything) I'd want a type error, not a runtime MatchError...)

Reply:
The warning is about the useless field. Also the translation could be made different, more like `val (x,y,z) = { current stuff; tupling }`. So the issue is not the side-effect, it's that a field is introduced. It doesn't warn here for locals, only class members.

Reply:
> It doesn't warn here for locals, only class members.

Ah, that's great.

So would those not warn using `locally`?  Mind adding those two variants in a `locally`, please?

Reply:
Sure, normally I use locally.

Reply:
I almost wonder if we should promote/suggest `locally` in the warning message...  WDYT?

Reply:
I wouldn't want this lint to get a bad rap when there's a way to encode the class initialisation, it just isn't well known.